common/display/displayqueue.cpp

```diff
@@ -149,27 +149,35 @@ bool DisplayQueue::ApplyPendingModeset(drmModeAtomicReqPtr property_set) {
   return true;
 }
 
-bool DisplayQueue::SetDpmsMode(uint32_t dpms_mode) {
+bool DisplayQueue::SetPowerMode(uint32_t power_mode) {
   ScopedSpinLock lock(spin_lock_);
-  if (dpms_mode_ == dpms_mode)
+
+  if (dpms_mode_ == power_mode)
     return true;
 
-  dpms_mode_ = dpms_mode;
-  if (dpms_mode_ == DRM_MODE_DPMS_OFF) {
+  if ((power_mode == kOff) || (power_mode == kDoze)) {
+    dpms_mode_ = DRM_MODE_DPMS_OFF;
+    Flush();
```

Comment:
User experience wont be as expected if we skip those queued flips. I hope that is the expected user behavior we get the new flip if there is any geometry change in the incoming layers. please correct me if am wrong